mountpoint-s3/src/inode/readdir.rs

```diff
@@ -25,13 +25,17 @@
 //!   inodes for them, achieving point 4. It also has a [ReaddirHandle::readd] method to handle
 //!   point 5.
 //! * [ReaddirIter] is an iterator over [ReaddirEntry]s, which are entries that may not yet have
-//!   inodes created for them. [ReaddirIter] merges together two streams, [RemoteIter] and
-//!   [LocalIter], to handle point 2. While merging, [ReaddirIter] also deduplicates the entries it
-//!   returns to handle point 1.
+//!   inodes created for them.
+//!   Addressing point 2, [ReaddirIter] merges together entries from two sources:
+//!   remotely from S3 using [RemoteIter] and locally from a snapshot of the parent's local children.
+//!   While merging, [ReaddirIter] also deduplicates the entries it returns to handle point 1.
```

Comment:
I can see the "deduplication" in the `ordered` implementation, but what about the `unordered`?

Reply:
The `unordered` implementation is dropping all local entries that have an entry returned of the same name by the remote iterator.

https://github.com/awslabs/mountpoint-s3/blob/57edf7714ac29d827f865dbe65a9d9f0f0f6d33b/mountpoint-s3/src/inode/readdir.rs#L561-L564

Reply:
Yes, but here we are talking about the deduplication in point 1: https://github.com/awslabs/mountpoint-s3/blob/57edf7714ac29d827f865dbe65a9d9f0f0f6d33b/mountpoint-s3/src/inode/readdir.rs#L6

I do not see where the `unordered` implementation is handling it. I may just be me, but it could merit a comment.